http/internal/logic/social/followLogic.go

```diff
@@ -24,7 +32,78 @@ func NewFollowLogic(ctx context.Context, svcCtx *svc.ServiceContext) *FollowLogi
 }
 
 func (l *FollowLogic) Follow(req *types.FollowRequest) (resp *types.FollowResponse, err error) {
-	// todo: add your logic here and delete this line
-
-	return
+	UserClaims, err := utils.ParseToken(req.Token, l.svcCtx.Config.JwtAuth.Secret)
+	if err != nil {
+		return nil, schema.ApiError{
+			StatusCode: 422,
+			Code:       42204,
+			Message:    "token解析失败",
+		}
+	}
+	_, err = l.svcCtx.UserRpc.QueryById(l.ctx, &user.QueryByIdRequest{
+		UserId: req.ToUserId,
+	})
+	if _, match := utils.MatchError(err, common.ErrUserNotFound); match {
+		return nil, schema.ApiError{
+			StatusCode: 422,
+			Code:       42202,
+			Message:    "关注的用户不存在",
+		}
+	}
+	if req.ActionType == Follow {
+		res, err := l.svcCtx.UserRpc.IsFollow(l.ctx, &user.IsFollowRequest{
+			UserId:   UserClaims.UserId,
+			TargetId: req.ToUserId,
+		})
+		isFollow := res.IsFollow
+		if isFollow {
+			return nil, schema.ApiError{
```

Comment:
In my opinion, there is no need to return this error since user's request already fulfilled, just return success

Reply:
😡 I don't think so!

Reply:
fine, I plus you.